web/src/message_view.ts

```diff
@@ -807,7 +807,8 @@ export let show = (raw_terms: NarrowTerm[], show_opts: ShowMessageViewOpts): voi
                     // If we switched narrow, there is nothing to do.
                     if (
                         msg_list.id !== message_lists.current?.id ||
-                        !id_info.first_unread_msg_id_pending_server_verification
+                        !id_info.first_unread_msg_id_pending_server_verification ||
+                        !msg_list.data.filter.is_conversation_view()
```

Comment:
How can this fail? We checked that at the top of this block, and checked the `message_list` ID hasn't changed -- is this a `with` thing? We should have a comment for what this is about.

Reply:
Yeah, modified to use an assert instead.